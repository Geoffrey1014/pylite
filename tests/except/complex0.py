def main():
    try:
        try:
            print 'main start'
            0/0
            print 'main end'
        except:
            print 'except'
    finally:
        print 'finally' 

print 'global start'
main()
print 'global end'
