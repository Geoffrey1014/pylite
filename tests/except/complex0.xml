lambda
  name __init__
  param_count 0
  local_count 0
  label_count 0
  insn_vector
    load_const c0
    store_global c1
    load_const c2
    print
    load_global c1
    call 0
    pop
    load_const c3
    print
    load_const c4
    return
  const_vector
    lambda
      name main
      param_count 0
      local_count 1
      label_count 8
      insn_vector
        label L0
        label L4
        load_const c0
        print
        load_const c1
        load_const c1
        div
        pop
        load_const c2
        print
        label L5
        jmp L7
        label L6
        pop
        load_const c3
        print
        label L7
        label L1
        jmp L3
        label L2
        store_local -1
        load_const c4
        print
        finally_exit -1
        label L3
        finally_call L2
        load_const c5
        return
      const_vector
        "main start"
        0.000000
        "main end"
        "except"
        "finally"
        None
      guard_vector
        guard
          start L0
          end L1
          handler L2
        guard
          start L4
          end L5
          handler L6
      line_number_map 4 4 4 4 5 5 5 5 6 6 4 4 4 4 8 8 4 4 4 4 4 10 10 4 4 4 1 1
    main
    "global start"
    "global end"
    None
  guard_vector
  line_number_map 1 1 12 12 13 13 13 14 14 1 1
