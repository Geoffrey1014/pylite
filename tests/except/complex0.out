global start
main start
except
finally
global end
